Patch to apply to Buildroot to add build for AP1302 legacy driver.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2022-05-05 12:07:00.966266161 +0200
+++ buildroot/package/Config.in	2022-05-05 12:19:55.911009193 +0200
@@ -8,6 +8,7 @@
 
 menu "Audio and video applications"
 	source "package/alsa-utils/Config.in"
+	source "package/ap1302-driver/Config.in"
 	source "package/aumix/Config.in"
 	source "package/bellagio/Config.in"
 	source "package/dvblast/Config.in"
Index: buildroot/package/ap1302-driver/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/ap1302-driver/Config.in	2022-05-05 12:24:01.106229536 +0200
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_AP1302_DRIVER
+	bool "AP1302 Linux driver"
+	depends on BR2_LINUX_KERNEL
+	help
+	  ap1302-driver - a kernel module to drive AP1302 ISP
+
+	  https://github.com/Avnet/ap1302-driver.git
+
+comment "ap1302-driver needs a Linux kernel >= ?? to be built"
+	depends on !BR2_LINUX_KERNEL
Index: buildroot/package/ap1302-driver/ap1302-driver.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/ap1302-driver/ap1302-driver.mk	2022-07-12 17:50:21.053478976 +0200
@@ -0,0 +1,33 @@
+################################################################################
+#
+# ap1302-driver
+#
+################################################################################
+
+#AP1302_DRIVER_VERSION = 97cc47cedbddadfd7074bae123b556db7065960f
+#AP1302_DRIVER_SITE = $(call github,Avnet,ap1302-driver,$(AP1302_DRIVER_VERSION))
+AP1302_DRIVER_VERSION = 59ec9b50782525522de90f596892b6547db5848a
+AP1302_DRIVER_SITE = $(call github,artemysfr,ap1302-legacy,$(AP1302_DRIVER_VERSION))
+AP1302_DRIVER_LICENSE = GPLv2
+AP1302_DRIVER_LICENSE_FILES = LICENSE
+
+AP1302_DRIVER_DEPENDENCIES = linux
+
+# cross compile environment for linux kernel modules:
+AP1302_DRIVER_MAKE_ENV = \
+	$(LINUX_MAKE_FLAGS) \
+	PREFIX=$(TARGET_DIR)/usr/local \
+	KERNELRELEASE=$(LINUX_VERSION_PROBED) \
+	KERNEL_DIR=$(LINUX_DIR) \
+	KERNEL_SRC=$(LINUX_DIR)
+
+define AP1302_DRIVER_BUILD_CMDS
+	$(MAKE) $(AP1302_DRIVER_MAKE_ENV) -C $(@D)
+endef
+
+define AP1302_DRIVER_INSTALL_TARGET_CMDS
+	$(MAKE) $(LINUX_MAKE_FLAGS) -C $(LINUX_DIR) M=$(@D) modules_install
+	$(INSTALL) -D -m 755 $(@D)/scripts/ap1302_regs.sh $(TARGET_DIR)/root/ap1302_regs.sh
+endef
+
+$(eval $(generic-package))
